sdks/java/io/singlestore/src/test/java/org/apache/beam/sdk/io/singlestore/SingleStoreIOIT.java

```diff
@@ -0,0 +1,267 @@
+import org.junit.BeforeClass;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+@RunWith(JUnit4.class)
+public class SingleStoreIOIT {
+
+  private static final String NAMESPACE = SingleStoreIOIT.class.getName();
+
+  private static final String DATABASE_NAME = "SingleStoreIOIT";
+
+  private static int numberOfRows;
+
+  private static String tableName;
+
+  private static String serverName;
+
+  private static String username;
+
+  private static String password;
+
+  private static Integer port;
+
+  private static SingleStoreIO.DataSourceConfiguration dataSourceConfiguration;
+
+  private static InfluxDBSettings settings;
+
+  @BeforeClass
+  public static void setup() {
+    SingleStoreIOTestPipelineOptions options;
+    try {
+      options = readIOTestPipelineOptions(SingleStoreIOTestPipelineOptions.class);
+    } catch (IllegalArgumentException e) {
+      options = null;
+    }
+    org.junit.Assume.assumeNotNull(options);
+
+    numberOfRows = options.getNumberOfRecords();
+    serverName = options.getSingleStoreServerName();
+    username = options.getSingleStoreUsername();
+    password = options.getSingleStorePassword();
+    port = options.getSingleStorePort();
+    tableName = DatabaseTestHelper.getTestTableName("IT");
+    dataSourceConfiguration =
+        SingleStoreIO.DataSourceConfiguration.create(serverName + ":" + port)
+            .withDatabase(DATABASE_NAME)
+            .withPassword(password)
+            .withUsername(username);
+    settings =
+        InfluxDBSettings.builder()
+            .withHost(options.getInfluxHost())
+            .withDatabase(options.getInfluxDatabase())
+            .withMeasurement(options.getInfluxMeasurement())
+            .get();
+  }
+
+  void createDatabaseIfNotExists() throws SQLException {
+    DataSource dataSource =
+        new SingleStoreDataSource(
+            String.format(
+                "jdbc:singlestore://%s:%d/?user=%s&password=%s&allowLocalInfile=TRUE",
+                serverName, port, username, password));
+    try (Connection conn = dataSource.getConnection();
+        Statement stmt = conn.createStatement()) {
+      stmt.executeQuery(String.format("CREATE DATABASE IF NOT EXISTS %s", DATABASE_NAME));
+    }
+  }
+
+  @Test
+  @Category(NeedsRunner.class)
+  public void testWriteThenRead() throws Exception {
+    createDatabaseIfNotExists();
+    DataSource dataSource =
+        new SingleStoreDataSource(
+            String.format(
+                "jdbc:singlestore://%s:%d/%s?user=%s&password=%s&allowLocalInfile=TRUE",
+                serverName, port, DATABASE_NAME, username, password));
+    DatabaseTestHelper.createTable(dataSource, tableName);
+    try {
+      PipelineResult writeResult = runWrite();
+      writeResult.waitUntilFinish();
+      PipelineResult readResult = runRead();
+      readResult.waitUntilFinish();
```

Comment:
We should assert the pipeline status for write and read is DONE; we may also assert the number of record we read is consistent with the number of record we have written (some basic validation). Looks like currently if the pipeline fails remotely the test will still pass.

Reply:
number of records is checked here 
```
  private void testReadResult(PCollection<TestRow> namesAndIds) {
    PAssert.thatSingleton(namesAndIds.apply("Count All", Count.globally()))
        .isEqualTo((long) numberOfRows);
```
(lines 248-250)

Will add asserts for pipeline status. Thanks!